utils/runtime-fuzzer/src/gear_calls.rs

```diff
@@ -0,0 +1,409 @@
+pub(crate) struct Config {
+    generators: Vec<RepeatedGenerator>,
+}
+
+impl Config {
+    pub fn new(generators: Vec<RepeatedGenerator>) -> Config {
+        Config { generators }
+    }
+
+    pub(crate) fn unstructured_size_hint(&self) -> usize {
+        self.generators
+            .iter()
+            .map(|generator| generator.amount * generator.generator.unstructured_size_hint())
+            .sum()
+    }
+}
+
+/// [`ExtrinsicGenerator`] that should be invoked [`RepeatedGenerator::amount`](`amount`) times
+/// by the [`GearCalls`] to generate [`RepeatedGenerator::amount`](`amount`) extrinsics.
+pub(crate) struct RepeatedGenerator {
+    pub amount: usize,
+    pub generator: Box<dyn ExtrinsicGenerator>,
+}
+
+impl RepeatedGenerator {
+    pub fn new(amount: usize, generator: Box<dyn ExtrinsicGenerator>) -> RepeatedGenerator {
+        RepeatedGenerator { amount, generator }
+    }
+}
+
+/// Trait that describes generator that can generate parameters to call extrinsics.
+pub(crate) trait ExtrinsicGenerator {
+    fn generate(
+        &self,
+        intermediate_data: &mut TempData,
+        unstructured: &mut Unstructured,
+    ) -> Result<Option<GearCall>>;
+
+    fn unstructured_size_hint(&self) -> usize;
+}
+
+/// Extrinsic generator that's capable of generating `UploadProgram` calls.
+pub(crate) struct UploadProgramGenerator {
+    pub gas: u64,
+    pub value: u128,
+    pub test_input_id: String,
+}
+
+impl ExtrinsicGenerator for UploadProgramGenerator {
+    fn generate(
+        &self,
+        intermediate_data: &mut TempData,
+        unstructured: &mut Unstructured,
+    ) -> Result<Option<GearCall>> {
+        log::trace!("New gear-wasm generation");
+        log::trace!("Random data before wasm gen {}", unstructured.len());
+
+        let code = gear_wasm_gen::generate_gear_program_code(
+            unstructured,
+            config(
+                &intermediate_data.uploaded_programs,
+                Some(format!(
+                    "Generated program from corpus - {}",
+                    &self.test_input_id
+                )),
+            ),
+        )?;
+        log::trace!("Random data after wasm gen {}", unstructured.len());
+        log::trace!("Code length {:?}", code.len());
+
+        let salt = arbitrary_salt(unstructured)?;
+        log::trace!("Random data after salt gen {}", unstructured.len());
+        log::trace!("Salt length {:?}", salt.len());
+
+        let payload = arbitrary_payload(unstructured)?;
+        log::trace!(
+            "Random data after payload (upload_program) gen {}",
+            unstructured.len()
+        );
+        log::trace!("Payload (upload_program) length {:?}", payload.len());
+
+        let program_id = ProgramId::generate(CodeId::generate(&code), &salt);
+
+        log::trace!("Generated code for program id - {program_id}");
+
+        intermediate_data.uploaded_programs.push(program_id);
```

Comment:
its not guaranteed to be successfully upload - gear::upload_program may fail

Reply:
This was the previous logics, so I decided to not change it in this PR as it may affect smth